persist/sqlite/consensus_refactored_test.go

```diff
@@ -2655,6 +2766,43 @@ func BenchmarkSiacoinOutputs(b *testing.B) {
 
 	n.mineTransactions(b, txn1)
 
+	// add a bunch of random outputs
```

Comment:
similar comment here, you should call `db.UnspentSiacoinOutputs(bm.addr, bm.offset, 100)` with a random offset and limit that you know is within bounds, so you can assert you always get `limit` number of unspent outputs

for `db.SiacoinElements` you have to take a similar approach where we test it with limits 10-100-1000 rather than always fetching a single element